### DISCUSSION TITLE
An error occurred packaging the executable with flet pack?

### DISCUSSION THREAD

**Discussion by Ben2022-cell** [flet-dev/flet#1028] [Q&A]
The [packaging-desktop-app](https://flet.dev/docs/guides/python/packaging-desktop-app) docs talk about packaging a desktop app into an executable using flet pack command as follows -

flet pack your_program.py

But wen i execute there code , there is presenting an error, see that:
$ flet pack jartual_frame_flet.py
Please install PyInstaller module to use flet pack command: cannot import name 'decode' from 'PyInstaller.utils.win32.versioninfo' (D:\Python39\lib\site-packages\PyInstaller\utils\win32\versioninfo.py)

the PyInstaller packing i has setup finish, and updated to the lastest version.

**Comment by ndonkoHenri** [comment]
What is shown when you do this:

```
pip show pyinstaller 
```
Then
```
pip show flet
```

And what is your OS?

**Reply by Ben2022-cell** [reply]
$ pip show pyinstaller
Name: pyinstaller
Version: 5.8.0
Summary: PyInstaller bundles a Python application and all its dependencies into a single package.
Home-page: https://www.pyinstaller.org/
Author: Hartmut Goebel, Giovanni Bajo, David Vierra, David Cortesi, Martin Zibricky
Author-email:
License: GPLv2-or-later with a special exception which allows to use PyInstaller to build and distribute non-free programs (including commercial ones)
Location: d:\python39\lib\site-packages
Requires: altgraph, pefile, pyinstaller-hooks-contrib, pywin32-ctypes, setuptools
Required-by:

$ pip show flet
Name: flet
Version: 0.4.0
Summary: Flet for Python - easily build interactive multi-platform apps in Python
Home-page:
Author: Appveyor Systems Inc.
Author-email: hello@flet.dev
License: MIT
Location: d:\python39\lib\site-packages
Requires: flet-core, httpx, oauthlib, packaging, watchdog, websocket-client, websockets
Required-by:

My OS is win10 21H1 version.

**Reply by iron3oxide** [reply]
Here's what I'm pretty sure is happening: in the flet pack command, flet tries to import `PyInstaller.__main__`, catches any ImportError that may happen and prints that out with the "install pyinstaller" prefix to let you know what probably went wrong, then exits.

Could you maybe open a REPL in the same virtual environment, execute `import PyInstaller.__main__` and paste the output here? That might give us a more verbose error trace.

**Reply by FeodorFitsner** [reply]
OK, the latest PyInstaller 5.8.0 broke the integration. Downgrade PyInstaller to 5.7.0 to make it work:

```
pip uninstall pyinstaller
pip install pyinstaller==5.7.0
```
